apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PgNumeric.cs

```diff
@@ -0,0 +1,481 @@
+﻿// Copyright 2022 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using System.Linq;
+using System.Numerics;
+using System.Text.RegularExpressions;
+
+namespace Google.Cloud.Spanner.V1
+{
+    /// <summary>
+    /// Representation of PostgreSQL numeric type which has max precision of 147455 and a max scale of 16383. 
+    /// </summary>
+    public struct PgNumeric : IEquatable<PgNumeric>, IComparable<PgNumeric>, IComparable
```

Comment:
I believe we can use `public readonly struct` which can help to optimize some cases.